pkg/api/steve/proxy/proxy.go

```diff
@@ -146,11 +147,24 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
 		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		var resource string
+		// Extract either a cluster-wide or namespaced resource from the URL.
+		switch len(splitPath) {
+		case 5:
```

Comment:
Do these numbers always work?

`/apis/management.cattle.io/v3/clusters` - the resource is element 4
`/api/v1/pods` - the resource is element 3

Can you add comments with example paths? Otherwise these just look like magic numbers.

You could use this path parser directly or as inspiration to make sure all cases are covered https://github.com/kubernetes/apiserver/blob/e78a7391c450c20ecbca04bdf9df6ef265689ddd/pkg/endpoints/request/requestinfo.go#L117-L247

Reply:
In case of `/api/v1/pods`, we want the function to return true - meaning the resource should be later handled by the auth function.

But we're specifically looking for exempt resources, and those are only composed of 4 or 6 elements, unless I am missing some corner case.

Reply:
Come to think of it, I wonder if we should change the name of the matcher function. It's not just about management resources, since core/v1 and any other resources could be its inputs.

Reply:
Could you still add a comment with an example?

Reply:
A correction: this code does **not** match core Kubernetes resources. It only matches management resources found in the allow-list.